src/models/AuxiliaryChain.ts

```diff
@@ -0,0 +1,94 @@
+import BigNumber from 'bignumber.js';
+import Comparable from '../observer/Comparable';
+
+/**
+ * Represents AuxiliaryChain model object.
+ */
+export default class AuxiliaryChain extends Comparable<AuxiliaryChain> {
+  public chainId: number;
+
+  public originChainName: string;
+
+  public ostGatewayAddress: string;
+
+  public ostCoGatewayAddress: string;
+
+  public anchorAddress: string;
+
+  public coAnchorAddress: string;
+
+  public lastProcessedBlockNumber: BigNumber;
+
+  public lastOriginBlockHeight: BigNumber;
+
+  public lastAuxiliaryBlockHeight: BigNumber;
+
+  public createdAt?: Date;
+
+  public updatedAt?: Date;
+
+  /**
+   * Constructor to set fields of AuxiliaryChain table.
+   *
+   * @param chainId Chain identifier.
+   * @param originChainName Name of the origin chain.
+   * @param ostGatewayAddress Gateway contract address.
+   * @param ostCoGatewayAddress CoGateway contract address.
+   * @param anchorAddress Anchor contract address.
+   * @param coAnchorAddress CoAnchor contract address.
+   * @param lastProcessedBlockNumber Last processed block number of that chain.
+   * @param lastOriginBlockHeight Latest orign chain block height.
+   * @param lastAuxiliaryBlockHeight Latest auxiliary chain block height.
+   * @param createdAt Time at which record is created.
+   * @param updatedAt Time at which record is updated.
+   */
+  public constructor(
+    chainId: number,
+    originChainName: string,
+    ostGatewayAddress: string,
+    ostCoGatewayAddress: string,
+    anchorAddress: string,
+    coAnchorAddress: string,
+    lastProcessedBlockNumber: BigNumber,
+    lastOriginBlockHeight: BigNumber,
+    lastAuxiliaryBlockHeight: BigNumber,
+    createdAt?: Date,
```

Comment:
The way I see model objects it should represent a complete db record. 
If we don't pass created_at and updated_at in constructor, in cases where we want to access **auxiliaryChain.created_at** and **auxiliaryChain.updated_at** it will not be accessible.
What do you think?

Reply:
One could unset and set those values after creation also. However, from the constructor perspective, I am not seeing a reason to have them as those are going to be only touched by repository classes.

Reply:
However, having as last arguments to constructor call is fine with me. Here, I would like to insist on making all, except chainId optional. To align with stake request model and repository.

Reply:
Changes done!